devicetypes/heltun/heltun-tps05-switch.src/heltun-tps05-switch.groovy

```diff
@@ -0,0 +1,667 @@
+	}
+}
+
+private channelNumber(String N) {
+	N.split(":")[-1] as Integer
+}
+
+def parse(String description) {
+	def result = null
+	def cmd = zwave.parse(description)
+	if (cmd) {result = zwaveEvent(cmd)}
+	return result
+}
+
+private void setState(value, endpoint = null) {
+	def map = [
+			encap(zwave.basicV1.basicSet(value: value), endpoint),
+			encap(zwave.switchBinaryV1.switchBinaryGet(), endpoint),
+	]
+	sendHubCommand(map, 500)
+}
+
+private encap(cmd, endpoint) {
+	if (endpoint) {
+		zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint:endpoint).encapsulate(cmd).format()
+	} else {
+		cmd.format()
+	}
+}
+
+def childOn(childId) {
+	setState(0xFF, channelNumber(childId))
+}
+
+def childOff(childId) {
+	setState(0, channelNumber(childId))
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd) {
+	def map = [:]
+	if (cmd.meterType == 1) {
+		if (cmd.scale == 0) {
+			map.name = "energy"
+			map.value = cmd.scaledMeterValue
+			map.unit = "kWh"
+			sendEvent(map)
+		} else if (cmd.scale == 2) {
+			map.name = "power"
+			map.value = Math.round(cmd.scaledMeterValue)
+			map.unit = "W"
+			sendEvent(map)
+		}
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.centralscenev1.CentralSceneNotification cmd) {
+	def numberOfButtons = state.numberOfButtons
+	def state
+	def buttonN
+	switch (cmd.keyAttributes as Integer) {
+		case 0: state = "pushed"; buttonN = cmd.sceneNumber; break
+		case 1: state = "up"; buttonN = cmd.sceneNumber; break
+		case 2: state = "held"; buttonN = cmd.sceneNumber; break
+	}
+	def buttonId = buttonN + numberOfButtons * 2
+	def child = childDevices?.find {channelNumber(it.deviceNetworkId) == buttonId }
+	child?.sendEvent([name: "button", value: state, data: [buttonNumber: 1], isStateChange: true])
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+	def numberOfButtons = state.numberOfButtons
+	def endPoint = cmd.sourceEndPoint.toInteger()
+	def encapsulatedCommand = cmd.encapsulatedCommand([0x32: 3, 0x25: 1, 0x20: 1])
+	def value = encapsulatedCommand.value
+	def childDevice = childDevices?.find {channelNumber(it.deviceNetworkId) == endPoint }
+	def corrRelCons = 0
+	def corRelParam = 11 + endPoint - numberOfButtons
+	if (endPoint in numberOfButtons..(2*numberOfButtons)) {
+		def param = parameterMap().find( {it.paramNum == corRelParam } ).name
+		def paramState = state."$param"
+		if (paramState) {
+			corrRelCons = paramState.value
+		}
+	}
+	if (childDevice) {
+		childDevice.sendEvent(name: "switch", value: value ? "on" : "off")
+		if (value) {
+			sendEvent(name: "switch", value: "on")
+			childDevice.sendEvent(name: "power", value: corrRelCons, unit: "W")
+		} else {
+			childDevice.sendEvent(name: "power", value: 0, unit: "W")
+			if (!childDevices.any { it.currentValue("switch") == "on" }) {
+				sendEvent(name: "switch", value: "off")
+			}
+		}
+	}
+}
```

Comment:
Can you have a look at how MultiChannelEncapsulation is handled in, for example, [zwave-multi-metering-switch](https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/a5269f8b26f6840cb3f1622936317d9ff45911c8/devicetypes/smartthings/zwave-multi-metering-switch.src/zwave-multi-metering-switch.groovy#L175)? I'm mostly referring to calling again `zwaveEvent`, instead resolving which event should be sent here, in MultiChannelEncap handler).

Reply:
At first, I did it that way, then I changed it as it was causing some delays

Reply:
What kind of delays? Perhaps there would be another way around them, as I really don't like this current solution.